az-digital/az_quickstart
# Closes #2911: PHP 8.2 compatibility.

## Description
- Upgrades PHP version to 8.2 in PHPStan, PHPCS, PHPUnit, DDEV, Lando, Probo, and GitHub actions configs
- Adds step to install yarn to Probo config since it is not pre-installed on new Probo image
- Fixes a few instances of deprecated code involving string interpolation and usage of the deprecated `utf8_encode`/`utf8_decode()` functions
- Updates entity API code in a few places to use `$entity->get('property')` instead of dynamic properties (`$entity->property`) since dynamic properties are deprecated in PHP 8.2 and PHPStan was complaining about them
- Adds PHP `XMLReader` class as a [Universal object crate](https://phpstan.org/config-reference#universal-object-crates) in PHPStan confg
- Adds patch for [smart_date PHP 8.2 compatibility issue](https://www.drupal.org/project/smart_date/issues/3381131)
- ~Adds patch for [seboettg/citeproc-php PHP 8.2 compatibility issue](https://github.com/seboettg/citeproc-php/issues/174) (Thanks @bberndt-uaz!)~ No longer needed

## Related issues




Closes #2911

## How to test


Link to Probo review site: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2915.probo.build

- General testing
- Check for deprecation errors in logs

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
